ui/lib/test-utils.tsx

```diff
@@ -0,0 +1,24 @@
+import { MuiThemeProvider } from "@material-ui/core";
+import { createMemoryHistory } from "history";
+import * as React from "react";
+import { Router } from "react-router-dom";
+import { ThemeProvider } from "styled-components";
+import theme, { muiTheme } from "./theme";
+
+export function withTheme(element) {
+  return (
+    <MuiThemeProvider theme={muiTheme}>
+      <ThemeProvider theme={theme}>{element}</ThemeProvider>
+    </MuiThemeProvider>
+  );
+}
+
+export function withContext(TestComponent, url: string) {
+  const history = createMemoryHistory();
+  history.push(url);
+  return (
```

Comment:
Oh cool, could we perhaps use Browser Router here as we do in the main app? I am thinking if we did that we wouldn't need to create the history etc.

Reply:
Yeah I see your point. This helped me https://testing-library.com/docs/example-react-router/#reducing-boilerplate (second point). I used to use React Testing Library also. Perfectly fine as is, I was just curious :)